### DISCUSSION TITLE
How to solve TypeError: Cannot read properties of undefined (reading 'startsWith')?

### DISCUSSION THREAD

**Discussion by delgesu1** [FlowiseAI/Flowise#267] [Q&A]
I am using the Conversational QA template and it was working before. Now I get a TypeError: Cannot read properties of undefined (reading 'startsWith') when I try to run the chatbot. Made new api keys for openai and pinecone, new pinecone index. Nothing seems to help. Thank you very much for any help!

**Comment by HenryHengZJ** [comment]
@delgesu1 are you on node >= 18.15.0?

No changes were made around `startsWith` it should work like before

**Reply by smxsm** [reply]
Make sure you upload a file for every document loader defined, or remove them

**Reply by PaulGrayAI** [reply]
I'm having the same issue, i've tried both to add files to the second document loader and also to remove the second document loader and still getting the same error?  Did you manage to solve? Thanks

**Comment by HMKahn006** [comment]
> I'm having the same issue, i've tried both to add files to the second document loader and also to remove the second document loader and still getting the same error? Did you manage to solve? Thanks

I'm in the same boat as Paul. I just posted a second thread in hopes of bringing more eyes to the conversation. Link here:
https://github.com/FlowiseAI/Flowise/discussions/2183

**Comment by acdewinter** [comment]
I ran into this error which seems to be a bug where the flow itself "corrupted" (using that term lightly because someone would need to review actual logs to see what is happening). The only solution was to rebuild the flow into a new flow (super annoying if you have a large one).

What I would like to see as a feature in general is to be able to save a node configuration (minus the inputs & outputs) under a "custom nodes" section so you can pull it out as configured in situations where you need to repeat a lot of things. Also duplicating a flow configuration would be great. (that might be there but I did not see it).
